src/routing/approved.rs

```diff
@@ -39,22 +38,83 @@ use bytes::Bytes;
 use ed25519_dalek::Verifier;
 use itertools::Itertools;
 use resource_proof::ResourceProof;
+use sn_data_types::{PublicKey as EndUserPK, Signature as EndUserSig};
 use sn_messaging::{
-    infrastructure::{
-        Error as InfrastructureError, GetSectionResponse, InfrastructureInformation,
-        Message as InfrastructureMessage,
+    network_info::{
+        Error as TargetSectionError, GetSectionResponse, Message as NetworkInfoMsg, NetworkInfo,
     },
     node::NodeMessage,
-    MessageType,
+    DstLocation, EndUser, MessageType, SrcLocation,
+};
+use std::{
+    cmp,
+    collections::{btree_map::Entry, BTreeMap},
+    net::SocketAddr,
+    slice, unimplemented,
 };
-use std::{cmp, net::SocketAddr, slice};
 use tokio::sync::mpsc;
 use xor_name::{Prefix, XorName};
 
 pub(crate) const RESOURCE_PROOF_DATA_SIZE: usize = 64;
 pub(crate) const RESOURCE_PROOF_DIFFICULTY: u8 = 2;
 const KEY_CACHE_SIZE: u8 = 5;
 
+type SocketId = XorName;
+
+struct EndUserRegistry {
```

Comment:
better put this `SocketId` and `EndUserRegistry` into a separate file?
this approved.rs is already too huge.

Reply:
Yep, also the aim (`file organisation`) before making this ready for PR. I should update the draft PR message with that into. (It was written in one of the commit msgs, f78a508)